app/app.go

```diff
@@ -1061,11 +1081,8 @@ func (d *Deployments) LookupDeployment(ctx context.Context,
 	}
 
 	for _, deployment := range list {
-		if deviceCount, err := d.db.DeviceCountByDeployment(ctx,
-			*deployment.Id); err != nil {
-			return nil, 0, errors.Wrap(err, "counting device deployments")
-		} else {
-			deployment.DeviceCount = deviceCount
+		if err := d.setDeploymentDeviceCountIfUnset(ctx, deployment); err != nil {
```

Comment:
if multiple instances of deployments service call `setDeploymentDeviceCountIfUnset` at the same time, isn't it possible that [$inc](https://github.com/mendersoftware/deployments/pull/429/commits/01aa2db54de6e0b7f64891d5f1f0bfb9b0de895e#diff-ea674f2e657b955202c2d74a4c7dec51R1533) will have unwanted outcome?

Reply:
that would be my concern as well. maybe make it idempotent after all, and just set a value to X?

Reply:
@merlin-northern @mchalski right, let's try a different approach: set if not set, inc (which is atomic) if already set. This should solve multiple instances working on the same deployment in a nice and clean way.